### DISCUSSION TITLE
My relative stuck in Donetsk, what should we do?

### DISCUSSION THREAD

**Discussion by andriilive** [community/community#34658] [Other Feature Feedback, Questions, & Ideas] [answered]
👋🏻 There

My far relative is stuck in Donetsk (Self called DNR now 😡).

I was showing him how I'm getting my job done on some of my side projects across Europe, delegating some smaller tasks to him for a smaller amount of pocket money (< 500 UAH / Mo) for his time spent.

Since Feb hi is hiding from the terrorist regime, sitting all the way home, with no chance to continue his studies or leave Donetsk. Our family is doing our best to get him out of there, but 4 now it seems impossible. So I'm trying to the minimum I can until he is locked: growing his skills.

The trouble is that all projects I maintain are private :( And **GitHub suspended the access to private repos from the 'DNR**':
https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#how-is-github-ensuring-that-folks-not-living-in-andor-having-professional-links-to-the-sanctioned-countries-and-territories-still-have-access-or-ability-to-appeal

Is there any way we still can collaborate at GitHub's private repos while he is still in the sanctioned region?

- He is Ukrainian and he has Ukrainian docs (Ukr + international passport - 2021 issued)
- I'm sending him money for his help at Ukrainian Card / Bank
- Hi is not participating/contributing to any terroristic shit

🙏🏻 help us

**Comment by airtower-luna** [comment]
Oh dear, that's a difficult situation! If anyone can help it's [GitHub support](https://support.github.com/request) (I assume this would fit under "account issues"), but I'm not sure how much they can do, having to follow government mandated sanctions. :confused: Us other users unfortunately can't do anything there, but I wish you the best! :blue_heart: :yellow_heart:

**Reply by andriilive** [reply]
I messaged GH support: attached transactions to verify his UKR card, bank account and late 2021 issued international passport (I don't think terrorist can get this easy 😀). But unf the restriction applies to the region, not by the persons residence, according to their's answer.

I'm anger, because the Github's **justice is broken here**. [F**ing terrorist Breivik seems a person, who CAN use GitHub](https://nymag.com/intelligencer/2014/02/anders-breivik-demanding-playstation-3-in-prison.html) && [Yandex acc](https://github.com/yandex) posting Russian propaganda at theirs homepage live is not restricted, as any other ru developer as far i know 😕

My personal opinion: in feature if the restrictions is US law based, the law need's to be finalised in cooperation (**and sure initiative from**) Ukrainian side. A real good example here can be a [MacPaw's ClearVPN case ](https://support.clearvpn.com/hc/uk/articles/5031145116060): they're giving free VPN access to every Ukrainian. To apply u should verify yourself through digital ```дія``` app or web, known as digital passport in Ukraine (It's government powered, and the above case means that they have a public API) 😀 With approach like that even Ukrainians in Donetsk, Lugansk or Russia are allowed to use their service, stay safe and follow real news. Cool, right?

You have no idea how many pro-ukrainian people (I've heard at least about ~45 students guys from my relatives circle) are now hiding at homes with no chance to go outside at Donetsk and Luhansk regions. Those people need at least a joy and a strict signal that they are not forgotten. 

But for now, i didn't gave up and finally found some working solutions, [described here](https://github.com/community/community/discussions/34658#discussioncomment-3891607) 
Keep calm 🤟🏻 and never give up

**Comment by bever1337** [comment, marked as answer]
Git is distributed by nature. Your colleague's changes can still be included in the upstream, GitHub repository even if they cannot directly interact with the GitHub service. To continue using GitHub, options might include:
1. Your machine's git server becomes the colleague's remote. Your colleague pushes to you, you push to GitHub.
2. Use git mirroring with another git service.

**Reply by andriilive** [reply]
> 1. Your machine's git server becomes the colleague's remote. Your colleague pushes to you, you push to GitHub.
> 2. Use git mirroring with another git service.

Yep was thinking in the same way and finally found a working solution. 
But instead of mines machine, I used **VM & remote development tools** 😀 

Started with a remote Ubuntu VM ad Digitalocean, then Coder and finally found and fell in love with [Gitpod](https://gitpod.io). 
- Free plan, is enough to play around. 
- It supports GitHub and GitLab integrations && makes all requests at the remote VM, hosted at EU or US 😉
- You can add SSH keys there
- It supports Vscode, JetBrains IDEA and PHPStorm
- Zero config to start
- Just ++2 files to your every repo and your project can easily run at Gitpods cloud VM. Docker like 
- I'm on Mac and my colleague is on Windows. I don't need to Docker projects anymore to make them easy to run for him. He don't need to WSL. I work with a local config just like before, and he connects to Gitpod powered remote Ubuntu with a  simple Dockers-like pre-setup, see [Gitpod's Docker](https://www.gitpod.io/docs/configure/workspaces/workspace-image). 
- Gitpod pre-builds the workspaces, so with a proper setup time from open in JB'S Gateway to running VM and cooked project (with all deps installed and resources build) is < 30sec for my cases

Here is a fast overview of some other things I found, tried && reached just in case someone else will need it:
- Mirroring is fine if u don't have a GH-based deploys setted up. Gitlab supports Github -> Gitlab mirroring 4free
- It's possible to configure git to automatically push and fetch **multiple remotes**. Works even with Github Desktop (I'm actually a git newbie so use that one mostly). The ```hub``` cli can be helpful for this approach.
- It's hard enough to create a side-by-side repos sync 😀 Possible solution can be ```Gitea``` on a VM somewhere && one of [these](https://github.com/search?q=gitea-sync) 
- It's hard enough to properly set up a remote Ubuntu VM with all features support, eg mine was: remote desktop, remote ssh, remote JetBrains.
- The remote **development is awesome**. U can set a env for a newbies with a few scripts. If they are Windows guys and you're not likely you will save some time for you and them dealing with a WSL or Docker. Other remote dev tools I've found: [Gitch](https://glitch.com/), [Coder 2](https://coder.com), [Coder 1 Digitalocean 1 Click Install](https://marketplace.digitalocean.com/apps/coder)
